sale_loyalty_order_suggestion_multi_product/wizard/sale_loyalty_reward_wizard.py

```diff
@@ -0,0 +1,100 @@
+
+    @api.depends("reward_ids", "selected_reward_id")
+    def _compute_multi_criteria(self):
+        self.multi_criteria = (
+            self.selected_reward_id.program_id.rule_ids.loyalty_criteria
+            == "multi_product"
+        )
+
+    def _compute_loyalty_rule_line_ids(self):
+        self.loyalty_rule_line_ids = None
+        # The products of the rule with criteria "multi_product" related to the
+        # product_id of the line are taken into consideration in case the line contains
+        # a product of the rules, otherwise the product_id will coincide with a reward
+        # product and in that case the products of the first rule "multi_product" are
+        # taken into consideration.
+        products = (
+            self.selected_reward_id.program_id.rule_ids.filtered(
+                lambda x: x.loyalty_criteria == "multi_product"
+                and self.product_id in x.loyalty_criteria_ids.product_ids
+            )[:1].loyalty_criteria_ids.product_ids
+            or self.selected_reward_id.program_id.rule_ids.filtered(
+                lambda x: x.loyalty_criteria == "multi_product"
+            )[:1].loyalty_criteria_ids.product_ids
+        )
+        if self.selected_reward_id and not self.applicable_program and products:
+            lines_vals = []
+            for record in products:
+                units_included = self.order_id.order_line.filtered(
+                    lambda x, record=record: x.product_id == record
+                    and not x.is_reward_line
+                ).product_uom_qty
+                lines_vals.append(
+                    Command.create(
+                        {
+                            "wizard_id": self.id,
+                            "product_id": record.id,
+                            "units_required": 1,
+                            "units_included": units_included or 0,
+                        },
+                    )
+                )
+            self.loyalty_rule_line_ids = lines_vals
+        else:
+            return super()._compute_loyalty_rule_line_ids()
+
+    def _compute_loyalty_rule_line_description(self):
+        self.loyalty_rule_line_description = False
+        products = (
+            self.selected_reward_id.program_id.rule_ids.loyalty_criteria_ids.product_ids
+        )
+        if self.selected_reward_id and not self.applicable_program and products:
+            if len(products) > 1:
+                product_names = products.mapped("name")
+                products_str = self.env._(
+                    "{products_list} {and_word} {last_product}"
+                ).format(
+                    products_list=", ".join(product_names[:-1]),
+                    and_word=self.env._("and"),
+                    last_product=product_names[-1],
+                )
+            self.loyalty_rule_line_description = self.env._(
+                "<b>* Required quantity:</b> 1 unit of %(products)s"
+            ) % {"products": products_str}
```

Comment:
In the translation function, you can pass the arguments as parameters.
```suggestion
            self.loyalty_rule_line_description = self.env._(
                "<b>* Required quantity:</b> 1 unit of %(products)s"
            ,
            products=products_str
            )
```

Reply:
all done, thanks!